internal/lightcone/nihility/goodnightandsleepwell/goodnightandsleepwell.go

```diff
@@ -0,0 +1,56 @@
+package goodnightandsleepwell
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Check key.Modifier = "good_night_and_sleep_well_check"
+)
+
+func init() {
+	lightcone.Register(key.GoodNightandSleepWell, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        4,
+		Path:          model.Path_NIHILITY,
+		Promotions:    promotions,
+	})
+
+	modifier.Register(Check, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeHitAll: onBeforeHitAll,
+		},
+	})
+}
+
+// For every debuff the target enemy has, the DMG dealt by the wearer increases by 12%/15%/18%/21%/24%,
+// stacking up to 3 time(s). This effect also applies to DoT.
+func Create(engine engine.Engine, owner key.TargetID, lc info.LightCone) {
+	stacks := float64(model.StatusType_STATUS_DEBUFF)
+	if stacks > 3 {
+		stacks = 3
+	}
+	amt := 0.09 + 0.03*float64(lc.Imposition)*stacks
```

Comment:
The stack check needs to happen inside the `onBeforeHitAll` listener. In the current state, a stat check here will only happen when the Light Cone is initially equipped (`Create` is called during sim startup when creating the new character, before any combat starts and when the enemies have not yet been instantiated).

This logic will not get a count of modifiers. `model.StatusType_STATUS_DEBUFF` is just a flag/enum value for how you can search for this data but not the data itself. To get the debuff status count of the target inside of `onbeforeHitAll` you would want to do the following:
```go
func onBeforeHitAll(mod *modifier.ModifierInstance, e event.HitStartEvent) {
  // how many debuffs are on the target being hit
  debuffCount := e.Hit.Defender.StatusCount(model.StatusType_STATUS_DEBUFF)
  ...
}
```